### PR TITLE
`RingBufferLogHandler.getView` needs to be a live `List` for use from `Jenkins.logRecords`

### PR DESCRIPTION
**Pull request by jglick**
https://github.com/jenkinsci/jenkins/pull/6018#issuecomment-990367907. Supersedes #6042.

### Proposed changelog entries

* Unreleased regression in system log display.

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by basil** [comment]
Looks like the `MemoryAssert` test failed on CI.

**Comment by timja** [comment]
> Looks like the `MemoryAssert` test failed on CI.

does `assumeTrue` not work in `RealJenkinsRule`? that `MemoryAssert` is supposed to be skipped on Java 11

File: test/src/test/java/jenkins/model/JenkinsLogRecordsTest.java
Comment (basil) [inline test/src/test/java/jenkins/model/JenkinsLogRecordsTest.java:37]: Testing of jenkinsci/jenkins-test-harness#358 failed, so I closed the PR.

```suggestion
        if (new VersionNumber(System.getProperty("java.specification.version")).isOlderThan(new VersionNumber("9"))) {
```

File: core/src/main/java/hudson/util/RingBufferLogHandler.java
Comment (mPokornyETM) [inline core/src/main/java/hudson/util/RingBufferLogHandler.java:106]: @timja  after this change is our log full with this messages.

image

 How can I deactivate it?
image

Comment (jglick) [reply core/src/main/java/hudson/util/RingBufferLogHandler.java:106]: This can happen if memory is full and garbage collection discards old messages. There is not a configuration option. Anyway I would not suggest using the GUI logger screen for real tracking; you can install the `support-core` plugin to get proper log files written to disk (and rotated).

Comment (mPokornyETM) [reply core/src/main/java/hudson/util/RingBufferLogHandler.java:106]: Thx, we will try it!

**Comment by basil** [comment]
Causes [JENKINS-68417](https://issues.jenkins.io/browse/JENKINS-68417).

**Pull request merged**
